base/boot.jl

```diff
@@ -218,19 +218,12 @@ end
 
 abstract DirectIndexString <: AbstractString
 
-immutable ASCIIString <: DirectIndexString
+immutable String <: AbstractString
     data::Array{UInt8,1}
-    ASCIIString(d::Array{UInt8,1}) = new(d)
+    String(d::Array{UInt8,1}) = new(d)
```

Comment:
Since this is no longer fresh and I'm going to have to rebase and rerun CI anyway, I'll include this change (and a few others) when I do.